src/test/java/com/shapesecurity/functional/data/ImmutableSetTest.java

```diff
@@ -107,4 +111,21 @@ public void containsKeyTest() {
         assertTrue(m.contains(2));
         assertFalse(m.contains(3));
     }
+
+
+    @Test
+    public void filterTest() {
+        ImmutableSet<String> expected = ImmutableSet.<String>emptyUsingEquality()
+            .put("key1")
+            .put("key2")
+            .put("keyx2")
+            .put("key3");
+        ImmutableSet<String> filteredSet = expected.filter(string -> string.endsWith("2"));
+        assertEquals(ImmutableSet.<String>emptyUsingEquality()
+                .put("key2")
+                .put("keyx2"),
+            filteredSet
```

Comment:
Let's also test that `hasher` remains unchanged. This will need at least two different hashers.